src/cluster/multi_step_cluster.cpp

```diff
@@ -63,36 +64,88 @@ vector<int> MultiStep::cluster(DatabaseFile &db, const vector<bool> *filter) {
 	return Util::Algo::greedy_vortex_cover(nb);
 }
 
+
+void MultiStep::steps(vector <vector<bool>> &reps, vector <vector <int>> &centroids, int count) {
+	count = count + 1;
+
+	if (reps.empty()) {
+		vector <bool> rep_c(rep_bitset(centroids.front()));
+		reps.push_back(rep_c);
+	}
+
+	else {
+		vector<bool> rep_c(rep_bitset(centroids.back(), &reps.back()));
+		reps.push_back(rep_c);
+
+		for (size_t i = 0; i < centroids.back().size(); ++i)
+			if (!reps.front()[i])
+				centroids.back()[i] = centroids.back()[centroids.front()[i]];
+	}
+
+	size_t n_rep_1 = count_if(reps.front().begin(), reps.front().end(), [](bool x) { return x; });
+	size_t n_rep_2 = count_if(reps.back().begin(), reps.back().end(), [](bool x) { return x; });
+
+	if (n_rep_1 == n_rep_2) {
+		message_stream << "Clustering step " << count << " complete. #Input sequences: " << centroids.front().size() << " #Clusters: " << n_rep_1 << endl;
+	}
+
+	else {
+		message_stream << "Clustering step " << count << " complete. #Input sequences: " << n_rep_1 << " #Clusters: " << n_rep_2 << endl;
+		
+		centroids.front().swap(centroids.back());
+		centroids.pop_back();
+
+		reps.front().swap(reps.back());
+		reps.pop_back();
+	}
+}
+		
 void MultiStep::run() {
 	if (config.database == "")
 		throw runtime_error("Missing parameter: database file (--db/-d)");
 	config.command = Config::makedb;
 	unique_ptr<DatabaseFile> db(DatabaseFile::auto_create_from_fasta());
 	const size_t seq_count = db->ref_header.sequences;
 
-	vector<int> centroid1(cluster(*db, nullptr));
-	vector<bool> rep1(rep_bitset(centroid1));
-	size_t n_rep1 = count_if(rep1.begin(), rep1.end(), [](bool x) { return x; });
-	message_stream << "Clustering step 1 complete. #Input sequences: " << centroid1.size() << " #Clusters: " << n_rep1 << endl;
+	vector<vector<int>> centroids;
+	vector<vector<bool>> reps;
+
+	
+	for (int i = 0; i <= config.cluster_steps.size(); i++) {
+		int c = i - 1;
 
-	config.sensitivity = Sensitivity::SENSITIVE;
-	vector<int> centroid2(cluster(*db, &rep1));
-	vector<bool> rep2(rep_bitset(centroid2, &rep1));
-	for (size_t i = 0; i < centroid2.size(); ++i)
-		if (!rep1[i])
-			centroid2[i] = centroid2[centroid1[i]];
-	message_stream << "Clustering step 2 complete. #Input sequences: " << n_rep1 << " #Clusters: " << count_if(rep2.begin(), rep2.end(), [](bool x) { return x; }) << endl;
+		if (i == 0) {
```

Comment:
Bitte gehe hier davon aus, dass auch der erste Schritt in `config.cluster_steps` hinterlegt ist.